RecoEgamma/EgammaElectronAlgos/src/ConversionFinder.cc

```diff
@@ -0,0 +1,300 @@
+
+    float d = sqrt(pow(xEl - xCand, 2) + pow(yEl - yCand, 2));
+    float dist = d - (rEl + rCand);
+
+    // this is equivalent to `1/tan(theta_1) - 1/tan(theta_2)` but requires less trigonometry
+    float dcot = ele.pz() / ele.pt() - track.get<Pz>() / track.get<Pt>();
+
+    //get the point of conversion
+    float xa1 = xEl + (xCand - xEl) * rEl / d;
+    float xa2 = xCand + (xEl - xCand) * rCand / d;
+    float ya1 = yEl + (yCand - yEl) * rEl / d;
+    float ya2 = yCand + (yEl - yCand) * rCand / d;
+
+    float x = .5f * (xa1 + xa2);
+    float y = .5f * (ya1 + ya2);
+    float rconv = sqrt(pow(x, 2) + pow(y, 2));
+    // The z-position of the conversion is unused, but here is how it could be computed if needed:
+    // float z = ele.dz() + rEl * ele.pz() * std::acos(1 - pow(rconv, 2) / (2. * pow(rEl, 2))) / ele.pt();
+
+    //now assign a sign to the radius of conversion
+    float tempsign = ele.px() * x + ele.py() * y;
+    tempsign = tempsign / std::abs(tempsign);
+    rconv = tempsign * rconv;
+
+    //return an instance of ConversionInfo, but with a NULL track refs
+    return {dist, dcot, rconv, std::nullopt, std::nullopt, -9999, -9999};
+  }
+
+  //------------------------------------------------------------------------------------
+
+  //takes in a vector of candidate conversion partners
+  //and arbitrates between them returning the one with the
+  //smallest R=sqrt(dist*dist + dcot*dcot)
+  ConversionInfo const& arbitrateConversionPartnersbyR(const std::vector<ConversionInfo>& convCandidates) {
+    ConversionInfo const* closestConversion = &convCandidates.front();
+
+    if (convCandidates.size() == 1)
+      return *closestConversion;
+
+    float R = pow(closestConversion->dist, 2) + pow(closestConversion->dcot, 2);
+
+    for (auto const& temp : convCandidates) {
+      float temp_R = pow(temp.dist, 2) + pow(temp.dcot, 2);
+      if (temp_R < R) {
+        R = temp_R;
+        closestConversion = &temp;
+      }
+    }
+
+    return *closestConversion;
+  }
+
+  //------------------------------------------------------------------------------------
+  ConversionInfo findBestConversionMatch(const std::vector<ConversionInfo>& v_convCandidates) {
+    using namespace std;
+
+    if (v_convCandidates.empty())
+      return ConversionInfo{-9999., -9999., -9999., std::nullopt, std::nullopt, -9999, -9999};
+
+    if (v_convCandidates.size() == 1)
+      return v_convCandidates.at(0);
+
+    vector<ConversionInfo> v_0;
+    vector<ConversionInfo> v_1;
+    vector<ConversionInfo> v_2;
+    vector<ConversionInfo> v_3;
+    //loop over the candidates
+    for (unsigned int i = 1; i < v_convCandidates.size(); i++) {
+      ConversionInfo temp = v_convCandidates.at(i);
+
+      if (temp.flag == 0) {
+        bool isConv = false;
+        if (std::abs(temp.dist) < 0.02 && std::abs(temp.dcot) < 0.02 && temp.deltaMissingHits < 3 &&
+            temp.radiusOfConversion > -2)
```

Comment:
`(temp.radiusOfConversion > -2)` is used everywhere, and it could be factorized out:
```
      if (temp.radiusOfConversion <= -2) continue;
```